src/platform/auth.ts

```diff
@@ -315,7 +315,7 @@ export function getUser(
 export const findUser = (
 	loginInfo: string,
 	tx?: Tx,
-): Promise<AnyObject | undefined> => {
+): Promise<Pick<User, typeof $select[number]> | undefined> => {
```

Comment:
Now that the result is typed, how about dropping the explicit return type declaration?

Reply:
No, please don't do this.

Having explicit return types helps tremendously when reading code, because you can just anchor your attention on these and connect the dots in your head without having to *read the function body*. I find extremely annoying that TS allows implicit returns.

You also get some practical benefits:

- You rule out compiler regressions on type inference/propagation
- You rule out errors during authoring -- you typically know what you *want* the function to return but might not notice that what you return is of incorrect type and the compiler will happily accept it.
- You help the compiler get type inference in the function body right, so you don't end up having to annotate intermediate results.

Reply:
@dfunckt just to be clear, are you fine with `Promise<Pick<User, typeof $select[number]> | undefined>`?

Reply:
Not sure what you're asking @thgreasi ?

If it's about whether I'd rather we had the return type on the signature, then yes, I'm strongly in favour of keeping the return type and not dropping it as you suggest in your first comment, for all the reasons in my previous commnet.

If it's about whether I have any issue with that specific return type, I must admit it's quite smarter than I feel comfortable with by referencing a constant in the function body, but I'm okay with it.